fe/fe-core/src/main/java/com/starrocks/replication/LakeReplicationJob.java

```diff
@@ -0,0 +1,160 @@
+            FileCacheInfo cacheInfo =
+                    FileCacheInfo.newBuilder().setEnableCache(false).setTtlSeconds(-1).setAsyncWriteBack(false).build();
+            // assume each shard group has only one shard
+            long shardGroupId = GlobalStateMgr.getCurrentState().getStarOSAgent().createFakeShardGroup();
+            Map<String, String> properties = new HashMap<>();
+            starOSAgent.createFakeShardWithId(pathInfo, cacheInfo, shardGroupId, properties, fakedShardId,
+                    WarehouseManager.DEFAULT_RESOURCE);
+            LOG.info("Created fake shard for storage volume: {}, shard id: {}, group id: {}",
+                    storageVolumeName, fakedShardId, shardGroupId);
+            return fakedShardId;
+        } catch (Throwable e) {
+            LOG.error("Failed to create shard for storage volume: " + storageVolume, e);
+            throw new RuntimeException("Failed to create shard for storage volume: " + storageVolume, e);
+        }
+    }
+
+    @Override
+    public void run() {
+        try {
+            if (super.getState().equals(ReplicationJobState.INITIALIZING)) {
+                beginTransaction();
+                sendReplicateLakeRemoteStorageTasks();
+                setState(ReplicationJobState.REPLICATING);
+            } else if (super.getState().equals(ReplicationJobState.REPLICATING)) {
+                if (isTransactionAborted()) {
+                    setState(ReplicationJobState.ABORTED);
+                } else if (isCrashRecovery()) {
+                    sendReplicateLakeRemoteStorageTasks();
+                    LOG.info("Lake replication job recovered, super.getState(): {}, database id: {}, table id: {}, transaction id: {}",
+                            super.getState(), super.getDatabaseId(), super.getTableId(), super.getTransactionId());
+                } else if (isAllTaskFinished()) {
+                    commitTransaction();
+                    setState(ReplicationJobState.COMMITTED);
+                }
+            }
+        } catch (Exception e) {
+            LOG.warn("Lake replication job exception, super.getState(): {}, database id: {}, table id: {}, transaction id: {}, ",
+                    super.getState(), super.getDatabaseId(), super.getTableId(), super.getTransactionId(), e);
+            abortTransaction(e.getMessage());
+            setState(ReplicationJobState.ABORTED);
+        }
+    }
+
+    private void sendReplicateLakeRemoteStorageTasks() throws Exception {
+        runningTasks.clear();
+        byte[] encryptionMeta = GlobalStateMgr.getCurrentState().getKeyMgr().getCurrentKEKAsEncryptionMeta();
+        for (PartitionInfo partitionInfo : super.getPartitionInfos().values()) {
+            for (IndexInfo indexInfo : partitionInfo.getIndexInfos().values()) {
+                for (TabletInfo tabletInfo : indexInfo.getTabletInfos().values()) {
+                    Long computeNodeId = GlobalStateMgr.getCurrentState().getWarehouseMgr()
```

Comment:
this is a multi-layer nested loop, avoid get the warehousemgr repeatedly in the inner most loop.